cdispyutils/hmac4/hmac4_auth_generator.py

```diff
@@ -157,6 +157,11 @@ def generate_presigned_url(
         ]
     )
     signature = generate_signature(signing_key.key, string_to_sign)
+
+    # escape special characters _after_ generating the signature, to match the
```

Comment:
this doesn't really require that it happens after, right? like this line could be right after `136`. it's just that the signature is calculated with literal spaces